### DISCUSSION TITLE
모듈이란 무엇인가요

### DISCUSSION THREAD

**Discussion by pipisebastian** [publdaze/js-study#101] [모듈]
- 모듈이란 무엇인가요
- 모듈의 장점에는 무엇이 있나요?

**Comment by publdaze** [comment]
> * 모듈이란 무엇인가요
- 재사용 가능한 코드 조각
- 일반적으로 파일 단위 분리
- 자신만의 파일(모듈) 스코프 가짐
- 캡슐화
> * 모듈의 장점에는 무엇이 있나요?
- 독자적인 스코프를 가지기 때문에 네임스페이스 오염 막을 수 있음
- 재사용 가능하고, 분리된 파일로 관리할 수 있어서 유지보수와 가독성에 좋음

**Comment by pipisebastian** [comment]
> * 모듈이란 무엇인가요
- 프로그램을 구성하는 요소이며, 모듈은 기능별로 분리되어, 파일 단위로 분리됩니다. (파일 스코프)
- 모듈 내용은 캡슐화되어 일반적으로는 디른 모듈에서 접근할 수 없습니다.

> * 모듈의 장점에는 무엇이 있나요?
- 기능 단위로 프로그램을 분리하며, 캡슐화를 통해 자원을 보호할 수 있으며, 재사용성이 좋습니다.
